### PR TITLE
Fixing xbox

### PR DESCRIPTION
**Pull request by zephirefaith**
## Motivation and Context

stretch_xbox_controller.py broke due to deleting previous SretchROSClient methods. Added those back and using StretchClient() for ensuring correct mode on the robot.

(also includes some SLAP stuff; sorry do not know how to cleanly bifurcate previous and new commits)

- [ ] Docs change / refactoring / dependency upgrade
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

# Checklist:

- [x] I have made changes to existing documentation where needed.
- [ ] I have added tests that show that the PR is functional.
- [x] I ran `tests/hw_manual_test.py` on hardware and observed that robot behavior looks normal.

### CODE REVIEW

File: projects/data_collection/collect_h5.py [hunk 1]
```diff
@@ -60,10 +60,10 @@ def toggle_episode(self):
         if not self._is_recording:
             self._is_recording = True
             self._keyframe_recorder.start_recording(self.task_name)
+            self._keyframe_recorder.save_frame()
+            print("Start frame saved")
         else:
             self._is_recording = False
```
Comment (cpaxton) [inline projects/data_collection/collect_h5.py:66]: this logic is really confusing, isn't this what was checked above?

File: projects/slap_manipulation/src/slap_manipulation/dataloaders/rlbench_loader.py [hunk 2]
```diff
@@ -186,7 +186,7 @@ def crop_around_voxel(self, xyz, rgb, voxel, crop_size):
         mask = np.linalg.norm(xyz - voxel, axis=1) < crop_size
         return xyz[mask], rgb[mask]
 
-    def downsample_point_cloud(self, xyz, rgb):
+    def shuffle_and_downsample_point_cloud(self, xyz, rgb):
```
Comment (cpaxton) [inline projects/slap_manipulation/src/slap_manipulation/dataloaders/rlbench_loader.py:189]: i like this rename

File: projects/slap_manipulation/src/slap_manipulation/dataloaders/rlbench_loader.py [hunk 3]
```diff
@@ -209,20 +209,20 @@ def remove_duplicate_points(self, xyz, rgb):
...
-                view_name = view + "_rgb"
-                rgb_ims.append(trial[view_name][idx])
-            f, axarr = plt.subplots(2, 2)
-            axarr[0, 0].imshow(rgb_ims[0])
-            axarr[0, 1].imshow(rgb_ims[1])
-            axarr[1, 0].imshow(rgb_ims[2])
-            axarr[1, 1].imshow(rgb_ims[3])
-            plt.show()
+        # TODO: remove deadcode
+        # debug_og = False
```
Comment (cpaxton) [inline projects/slap_manipulation/src/slap_manipulation/dataloaders/rlbench_loader.py:213]: make this a func param

File: src/home_robot_hw/home_robot_hw/ros/recorder.py [hunk 4]
```diff
@@ -21,11 +22,7 @@ class Recorder(object):
     def __init__(self, filename, start_recording=False, model=None, robot=None):
         """Collect information"""
         print("Connecting to robot environment...")
-        self.robot = (
-            HelloStretchROSInterface(visualize_planner=False, model=model)
-            if robot is None
-            else robot
-        )
+        self.robot = StretchManipulationEnv(init_cameras=True)
```
Comment (cpaxton) [inline src/home_robot_hw/home_robot_hw/ros/recorder.py:25]: should this be the new client by @exhaustin

**Review by cpaxton — COMMENTED** [review]
minor changes

File: src/home_robot_hw/home_robot_hw/ros/recorder.py (hunk 4 above)
Comment (zephirefaith) [reply src/home_robot_hw/home_robot_hw/ros/recorder.py:25]: I think so...but I do not want to handle that change in this PR. I can make that change in my SLAP PR.

File: projects/data_collection/collect_h5.py (hunk 1 above)
Comment (cpaxton) [reply projects/data_collection/collect_h5.py:66]: never mind its fine

File: src/home_robot_hw/home_robot_hw/ros/recorder.py (hunk 4 above)
Comment (cpaxton) [reply src/home_robot_hw/home_robot_hw/ros/recorder.py:25]: file an issue

**Review by cpaxton — APPROVED** [review]
lgtm

**Pull request merged**